### PR TITLE
fix(TDP-4441): Add support for duplicated field when converting DP to Avro

### PR DESCRIPTION
**Pull request by fhuaulme**
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-4441

**Please check if the PR fulfills these requirements**
- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [x] The new code does not introduce new technical issues (sonar / eslint)
- [ ] Functional tests have been performed
- [ ] Docker configuration files for config-std or config-cloud profiles are impacted

**Please check the browsers you've tested on**
- [ ] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !
- [x] No, and no need to (backend changes only)

* Remove not used classes ColumnMetadataWrapper & RowMetadataWrapper.

### CODE REVIEW

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/699/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/699/)

File: dataprep-backend-common/src/main/java/org/talend/dataprep/api/dataset/row/RowMetadataUtils.java [hunk 1]
```diff
@@ -61,7 +55,22 @@ public static Schema toSchema(List<ColumnMetadata> columns) {
...
+                        // Modify column name
+                        final int suffix = uniqueSuffixes.get(columnMetadata.getName());
+                        uniqueSuffixes.put(columnMetadata.getName(), suffix + 1);
+                        columnMetadata.setName(columnMetadata.getName() + '_' + suffix);
+                    } else {
+                        // Don't modify column name
+                        uniqueSuffixes.put(columnMetadata.getName(), 1);
+                    }
+                })
+                .collect(Collectors.toList());
```
Comment (hmdebenque) [inline dataprep-backend-common/src/main/java/org/talend/dataprep/api/dataset/row/RowMetadataUtils.java]: why collect and not join the two streams?

File: dataprep-backend-common/src/main/java/org/talend/dataprep/api/dataset/row/RowMetadataUtils.java [hunk 2]
```diff
@@ -97,8 +106,11 @@ public static String toAvroFieldName(ColumnMetadata column) {
             if (i == 0) {
                 if (!Character.isLetter(currentChar)) {
                     columnName.append(DATAPREP_FIELD_PREFIX);
+                } else if (!Character.isJavaIdentifierPart(currentChar)) {
```
Comment (hmdebenque) [inline dataprep-backend-common/src/main/java/org/talend/dataprep/api/dataset/row/RowMetadataUtils.java]: I do not see the correlation with Avro specification on names: https://avro.apache.org/docs/current/spec.html#names

File: dataprep-backend-common/src/main/java/org/talend/dataprep/api/dataset/row/RowMetadataUtils.java [hunk 3]
```diff
@@ -61,7 +55,22 @@ public static Schema toSchema(List<ColumnMetadata> columns) {
 
     public static Schema toSchema(String name, List<ColumnMetadata> columns) {
 
-        List<Schema.Field> fields = columns.stream() //
+        final Map<String, Integer> uniqueSuffixes = new HashMap<>();
+        final List<ColumnMetadata> uniqueColumns = columns.stream() //
+                .peek(columnMetadata -> {
+                    if (uniqueSuffixes.containsKey(columnMetadata.getName())) {
```
[the remark this answers is not in the document]
Comment (fhuaulme) [reply dataprep-backend-common/src/main/java/org/talend/dataprep/api/dataset/row/RowMetadataUtils.java]: :+1:

File: dataprep-backend-common/src/main/java/org/talend/dataprep/api/dataset/row/RowMetadataUtils.java (hunk 2 above)
Comment (fhuaulme) [reply dataprep-backend-common/src/main/java/org/talend/dataprep/api/dataset/row/RowMetadataUtils.java]: :-1:

File: dataprep-backend-common/src/main/java/org/talend/dataprep/api/dataset/row/RowMetadataUtils.java (hunk 1 above)
Comment (fhuaulme) [reply dataprep-backend-common/src/main/java/org/talend/dataprep/api/dataset/row/RowMetadataUtils.java]: :+1:

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/701/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/701/)

**Pull request merged**